packages/odata-service-inquirer/src/prompts/connectionValidator.ts

```diff
@@ -117,19 +154,107 @@ export class ConnectionValidator {
+     *
+     * @param url the service or system url
+     * @param ignoreCertError if true the config will be set to ignore cert errors
+     * @param username provided for basic authentication
+     * @param password provided for basic authentication
+     * @returns the axios configuration object
+     */
+    private createAxiosConfig(
+        url: URL,
+        ignoreCertError: boolean,
+        username: string | undefined,
+        password: string | undefined
+    ) {
+        let axiosConfig: AxiosRequestConfig & ProviderConfiguration = {
+            params: Object.fromEntries(url.searchParams),
+            ignoreCertErrors: ignoreCertError,
+            cookies: '',
+            baseURL: url.origin
+        };
+
+        if (username && password) {
+            axiosConfig = Object.assign(axiosConfig, {
+                auth: {
+                    username,
+                    password
+                }
+            });
+        }
+        return axiosConfig;
+    }
+
+    /**
+     * Create the connection for a service url. The base url should be provided with the axios config property.
+     *
+     * @param axiosConfig the axios request configuration
+     * @param servicePath the service path without the origin
+     */
+    private async createServiceConnection(axiosConfig: AxiosRequestConfig, servicePath: string) {
+        this._axiosConfig = axiosConfig;
+        this._serviceProvider = create(this._axiosConfig);
+        this._odataService = this._serviceProvider.service(servicePath);
+        LoggerHelper.attachAxiosLogger(this._serviceProvider.interceptors);
+        await this._odataService.get('');
+    }
+
+    /**
+     * Create the connection for a system url. The system url should be provided as a base url axios config property.
+     *
+     * @param axiosConfig the axios request configuration
+     * @param odataVersion the odata version to restrict the catalog requests if only a specific version is required
+     */
+    private async createSystemConnection(axiosConfig: AxiosRequestConfig, odataVersion?: ODataVersion) {
+        this._axiosConfig = axiosConfig;
+        this._serviceProvider = createForAbap(this._axiosConfig);
+        LoggerHelper.attachAxiosLogger(this._serviceProvider.interceptors);
+
+        if (!odataVersion || odataVersion === ODataVersion.v2) {
+            this._catalogV2 = (this._serviceProvider as AbapServiceProvider).catalog(ODataVersion.v2);
+        }
+        if (!odataVersion || odataVersion === ODataVersion.v4) {
+            this._catalogV4 = (this._serviceProvider as AbapServiceProvider).catalog(ODataVersion.v4);
+        }
+        try {
+            this._catalogV2 ? await this._catalogV2.listServices() : await this._catalogV4.listServices();
+        } catch (error) {
+            // We will try the v4 catalog if v2 returns a 404
+            if ((error as AxiosError).response?.status === 404 && this._catalogV4) {
```

Comment:
What's the reason for only doing this for a 404? 
I have a system I do not have auth to access the V4 catalog and it returns a 403.

Reply:
403 is handled elsewhere (auth handling). This is the "ping" to determine the service catalog availability. Some systems do not have v2 catalog enabled (which should return a 404) and so we "fallback" to v4 catalog in this specific case.

Reply:
Ok thanks

Reply:
Thanks for including this.